services/workers/src/main/java/org/sagebionetworks/object/snapshot/worker/utils/PrincipalObjectRecordWriter.java

```diff
@@ -88,21 +100,32 @@ public void buildAndWriteRecord(ChangeMessage message) throws IOException {
 	}
 
 	/**
-	 * Log all teams that this principal belongs to
-	 * @param principalId
-	 * @param limit - the max number of teams will be written in to a log file at a time
+	 * Log all members that belongs to this group
+	 * 
+	 * @param groupId
+	 * @param limit - the max number of members will be written in to a log file at a time
 	 * @param timestamp - the timestamp of the change message
 	 * @throws IOException 
 	 */
-	public void captureAllTeams(Long principalId, long limit, long timestamp) throws IOException {
-		long offset = 0;
-		long numberOfTeams = teamDAO.getCountForMember(principalId.toString());
-		
-		while (offset < numberOfTeams) {
-			List<Team> teams = teamDAO.getForMemberInRange(principalId.toString(), limit, offset);
+	public void captureAllMembers(String groupId, int limit, long timestamp) throws IOException {
+		int offset = 0;
+		List<UserGroup> members = groupMembersDAO.getMembers(groupId);
+
+		while (offset < members.size()) {
+			List<UserGroup> membersToWrite = members.subList(offset, Math.min(offset+limit, members.size()));
 			List<ObjectRecord> records = new ArrayList<ObjectRecord>();
-			for (Team team : teams) {
-				TeamMember teamMember = teamDAO.getMember(team.getId(), principalId.toString());
+			for (UserGroup member : membersToWrite) {
+				TeamMember teamMember = null;
+				if (BOOTSTRAP_PRINCIPALS.contains(groupId)) {
+					teamMember = new TeamMember();
+					teamMember.setTeamId(groupId);
+					UserGroupHeader ugh = new UserGroupHeader();
+					ugh.setOwnerId(member.getId());
+					teamMember.setMember(ugh);
+					teamMember.setIsAdmin(false);
+				} else {
+					teamMember = teamDAO.getMember(groupId, member.getId());
+				}
```

Comment:
remove limit and offset, capture team members the same way group members are captured